LuckyDragon/app/src/main/java/com/example/luckydragon/User.java

```diff
@@ -113,12 +114,12 @@ public void save() {
         map.put("isAdmin", isAdmin());
 
         if (isOrganizer()) {
-            map.put("facility", organizer.getFacility());
+            map.put("facility", nonNull(organizer.getFacility()) ? organizer.getFacility() : null);
         }
 
         map.put("name", name);
-        map.put("email", email);
-        map.put("phoneNumber", phoneNumber);
+        map.put("email", nonNull(email) ? email : null);
+        map.put("phoneNumber", nonNull(phoneNumber) ? phoneNumber : null);
```

Comment:
redundant?

Reply:
Ah ok, are you sure that this still doesn't save the null value to firestore? I thought you would need to do something like:
`if (nonNull(email)) { map.put("email", email) }`

Reply:
Wasn't sure if it would be better to omit empty fields from the database entirely or to just set them to null. This will save the null value to firestore.

If we want to omit empty fields from the database entirely, if a user changes their phone number from some number to empty then we would have to remove the phone number item from the database. Setting them to null might be a little simpler to implement and makes it explicit that those fields may exist but are currently empty.

Reply:
!email.isEmpty() ? email : null will error if email is null. I made it nonNull(email) && !email.isEmpty ? email : null because it is a good failsafe to have empty fields set to null if they somehow sneak through